Sources/StreamChat/WebSocketClient/EventMiddlewares/MemberEventMiddleware.swift

```diff
@@ -13,7 +13,19 @@ struct MemberEventMiddleware: EventMiddleware {
                 try session.saveMember(payload: event.member, channelId: event.cid)
                 
             case let event as MemberAddedEventDTO:
-                try session.saveMember(payload: event.member, channelId: event.cid)
+                let member = try session.saveMember(payload: event.member, channelId: event.cid)
+                
+                // If there are any `MemberListQuery`s observing this Channel
+                // without any filters (so the query observes all members)
+                // the new Member should be linked to them too
+                // so `MemberListController` works as expected
+                // To make it work with queries with filters, we need to mirror `ChannelListController` logic
+                // `shouldListUpdatedChannel` and such
+                if let channel = session.channel(cid: event.cid) {
```

Comment:
Should we even save a member if the channel does not exist locally?

Reply:
Channel data will be saved in another middleware

Reply:
Yeah, but the event holds only `cid` 🙂 If we get this event without having this channel locally we will save a member. 

My suggestion is to `guard let channel = ... else { break }` before `let member = ...; channel.members.insert(member)`

Reply:
on another thought, you can have a `MemberListController` without having the channel data saved?

Reply:
Hm, good point 👍 The member indeed can be useful for `MemberListController` but on the other hand, it can break the pagination if the member delivered with the event is outside the fetched range

Reply:
since we are moving towards the state when search results are not persisted checking the channel existence sounds a bit safer

Reply:
alright, fair point